pkgs/development/ruby-modules/gem/default.nix

```diff
@@ -142,6 +142,10 @@ stdenv.mkDerivation ((builtins.removeAttrs attrs ["source"]) // {
       gempkg=$(echo "$output" | grep -oP 'File: \K(.*)')
 
       echo "gem package built: $gempkg"
+    elif [[ "$type" == "git" ]]; then
+      git init
+      rm -rf .git/logs/ .git/hooks/ .git/index .git/FETCH_HEAD .git/ORIG_HEAD .git/refs/remotes/origin/HEAD .git/config
+      git add .
```

Comment:
what is the purpose of having the files in the staging area?

Reply:
Adding this at build time can be useful for `git ls-files` to work.